README.md

```diff
@@ -131,23 +115,23 @@ This gets really messy with callbacks. But with `Deferred`, it's just:
 
 ```swift
 // Let's use type annotations to make it easier to see what's going on here.
-let friends: Deferred<[Friend]> = AsynchronousFriends(forUser: jimbob)
-let names: Deferred<[Name]> = friends.flatMap { friends in
+let friends: Deferred<[Friend]> = fetchFriends(forUser: jimbob)
+let names: Future<[Name]> = friends.flatMap { (friends: [Friend]) -> Future<[Name]> in
     // fork: get an array of not-yet-determined names
     let names: [Deferred<Name>] = friends.map { AsynchronousCall(.Name, friend: $0) }
 
     // join: get a not-yet-determined array of now-determined names
-    let allNames: Deferred<[Name]> = all(names)
-    return allNames
+    return names.joinedValues
 }
-names.upon { names: [Name] in
+
+names.upon { (names: [Name]) in
     // names has been determined - use it!
     dataSource.array = names
     tableView.reloadData()
 }
 ```
 
-## Tasks
+## Basic Tasks
 
 ### Vending a Future
```

Comment:
I like that. Maybe even go with "Vending a Future Value" to stick with the
"this is a value that hasn't quite arrived yet". Then we get future value
vs future type verbal distinction, even absent any fancy markup.

--
Jeremy W. Sherman <jeremy@bignerdranch.com>

2016-04-12 13:58 GMT-04:00 Zachary Waldowski <notifications@github.com>:

> In README.md
> <https://github.com/bignerdranch/Deferred/pull/94#discussion_r59422781>:
>
> >      // names has been determined - use it!
> >      dataSource.array = names
> >      tableView.reloadData()
> >  }
> >  ```
> >
> > -## Tasks
> > +## Basic Tasks
> >
> >  ### Vending a Future
>
> I think we would be well served by
>
>    - "Future" (the idea)
>    - "the Future type" or "the Future<Value> type"
>    - "Deferred" (the framework)
>    - "the Deferred type" or "the Deferred<Value> type"
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/bignerdranch/Deferred/pull/94/files/b56bbd18e89e5b391461bb520073b5f226dfdb9a#r59422781>
>